openedx/core/djangoapps/theming/finders.py

```diff
@@ -0,0 +1,65 @@
+"""
+Static file finders for Django.
+https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-STATICFILES_FINDERS
+Yes, this interface is private and undocumented, but we need to access it anyway.
+"""
+from path import Path
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.contrib.staticfiles import utils
+from django.contrib.staticfiles.finders import BaseFinder
+from openedx.core.djangoapps.theming.storage import CachedComprehensiveThemingStorage
+
+
+class ComprehensiveThemeFinder(BaseFinder):
+    """
+    A static files finder that searches the active comprehensive theme
+    for locate files. If the ``COMP_THEME_DIR`` setting is unset, or the
+    ``COMP_THEME_DIR`` does not exist on the file system, this finder will
+    never find any files.
+    """
+    def __init__(self, *args, **kwargs):
+        COMP_THEME_DIR = getattr(settings, "COMP_THEME_DIR", "")  # pylint: disable=invalid-name
+        if not COMP_THEME_DIR:
+            self.storage = None
+            return
+
+        if not isinstance(settings.COMP_THEME_DIR, basestring):
```

Comment:
Ah, got it. Fixed.

And yes, I think it's wise to account for the possibility that the setting might be missing, since we want to support the use case of *not* having a comprehensive theme enabled at all.